diesel_infer_schema/src/inference.rs

```diff
@@ -130,3 +130,18 @@ pub fn get_primary_keys(
         Ok(primary_keys)
     }
 }
+
+pub fn load_foreign_key_constraints(database_url: &str, schema_name: Option<&str>)
+    -> Result<Vec<ForeignKeyConstraint>, Box<Error>>
+{
+    let connection = try!(establish_connection(database_url));
+
+    match connection {
+        #[cfg(feature = "sqlite")]
+        InferConnection::Sqlite(c) => ::sqlite::load_foreign_key_constraints(&c, schema_name),
+        #[cfg(feature = "postgres")]
+        InferConnection::Pg(c) => ::information_schema::load_foreign_key_constraints(&c, schema_name).map_err(Into::into),
```

Comment:
It will also work for SQL Server, Oracle, Spanner, and any other database which correctly implements the `information_schema` ANSI standard.